cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/PersistedMessageCallback.java

```diff
@@ -120,10 +112,7 @@ private void recordLastOffset() {
       if (lastOffset >= 0) {
         offsets.put(lastTopicPartition, lastOffset);
       }
-      if (messageCount.get() >= persistThreshold) {
```

Comment:
@gsps1 Can you confirm that the offset persisting and writing metrics to HBase happen in the same transaction?